### PR TITLE
[CRIMAP-532] Introduce RoleChangedNotifier

### PR DESCRIPTION
**Pull request by mseedat-moj**
## Description of change
Notifies all admins whenever a user role is changed.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-532

#### Email contents:
Subject:
>((user_name)) is now a ((role_name)) for "Review criminal legal aid applications"

Body:
```
Role changed for ((user_name))‘s Review criminal legal aid applications account
If this was an unauthorised action please contact ((onboarding_email)) immediately.

Go to https://staging.review-criminal-legal-aid.service.justice.gov.uk/admin/manage_users to manage user roles.

From,
The Crime Review team

```

### CODE REVIEW

File: app/notifiers/slack.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Slack
+  attr_reader :token, :channel
+
+  def initialize(channel)
+    @token = Rails.configuration.x.admin.slack_token
+    @channel = channel
+  end
+
+  def post_message(_text); end
+end
```
Comment (mseedat-moj) [inline app/notifiers/slack.rb:10]: Any interest in pursuing Slack App setup/implementation? Otherwise will remove.

**Review by timpeat — APPROVED** [review]
Looks great! Thank you.

**Review by arthurashman — APPROVED** [review]
Nice 👍

File: app/mailers/notify_mailer.rb [hunk 2]
```diff
@@ -48,6 +48,19 @@ def revive_account_email(email)
...
   end
 
+  def role_changed_email(emails, user)
+    set_template(:role_changed_email)
+    set_email_reply_to(:onboarding_reply_to_address)
+
+    set_personalisation(
+      user_name: user.name,
+      role_name: user.role.humanize,
+      onboarding_email: Rails.configuration.x.admin.onboarding_email
```
Comment (willmcb) [inline app/mailers/notify_mailer.rb:58]: I am wondering if this should be the intercept email address rather than this one?

**Review by willmcb — APPROVED** [review]
Nice one 👍 Happy for this to go - small comment that we can talk about as a group tomorrow.

**Pull request merged**